analyzers/src/SonarAnalyzer.Common/Metrics/MetricsBase.cs

```diff
@@ -25,28 +25,29 @@ public abstract class MetricsBase
         protected const string InitializationErrorTextPattern = "The input tree is not of the expected language.";
 
         private readonly SyntaxTree tree;
+        private readonly string filePath;
 
         internal static readonly string[] LineTerminators = { "\r\n", "\n", "\r" };
 
         protected MetricsBase(SyntaxTree tree)
         {
             this.tree = tree;
+            filePath = tree.GetRoot().GetMappedFilePathFromRoot();
         }
 
-        public int LineCount =>
-            tree.GetText().Lines.Count;
-
         public abstract ImmutableArray<int> ExecutableLines { get; }
 
         public ISet<int> CodeLines =>
             tree.GetRoot()
                 .DescendantTokens()
-                .Where(token => !IsEndOfFile(token))
+                .Where(x => !IsEndOfFile(x))
+                .Select(x => x.GetLocation().GetMappedLineSpan())
+                .Where(IsInSameFile)
                 .SelectMany(
-                    t =>
+                    x =>
                     {
-                        var start = t.GetLocation().GetLineSpan().StartLinePosition.Line + 1;
-                        var end = t.GetLocation().GetLineSpan().EndLinePosition.Line + 1;
+                        var start = x.StartLinePosition.Line + 1;
+                        var end = x.EndLinePosition.Line + 1;
```

Comment:
Educational: why +1?

Reply:
The line span location is 0-based while the SonarQube one starts with 1.